File: modules/native-crash/src/NativeCrashModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class NativeCrashModule extends NativeModule {
  crash(message: string): void;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<NativeCrashModule>('NativeCrash');

File: modules/native-crash/android/src/main/java/com/west2online/nativecrash/NativeCrashModule.kt
package com.west2online.nativecrash

import expo.modules.BuildConfig
import expo.modules.kotlin.functions.Queues
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeCrashModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("NativeCrash")

        Function("crash") { message: String ->
            throw RuntimeException(message)
            return@Function
        }
    }
}
